### PR TITLE
Telemetry / Upgrade Responder Integration

### PR DESCRIPTION
**Pull request by irq0**
Summary:
- Start background thread that periodically POSTs to the s3gw upgrade responder URL
- Parses some of the response to get the list of available versions
- Add status page showing service status, available versions and the request data

Issue: https://github.com/aquarist-labs/s3gw/issues/423

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

### CODE REVIEW

**Comment by irq0** [comment]
back to draft to incorporate feedback from https://github.com/aquarist-labs/s3gw/issues/202

File: src/rgw/rgw_status_page_telemetry.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=2 sw=2 expandtab ft=cpp
+
+/*
+ * Ceph - scalable distributed file system
+ * SFS SAL implementation
+ *
+ * Copyright (C) 2022 SUSE LLC
```
Comment (votdev) [inline src/rgw/rgw_status_page_telemetry.h]:
```suggestion
 * Copyright (C) 2023 SUSE LLC
```

File: src/rgw/rgw_status_page_telemetry.cc [hunk 2]
```diff
@@ -0,0 +1,108 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=2 sw=2 expandtab ft=cpp
+
+/*
+ * Ceph - scalable distributed file system
+ * SFS SAL implementation
+ *
+ * Copyright (C) 2022 SUSE LLC
```
Comment (votdev) [inline src/rgw/rgw_status_page_telemetry.cc]:
```suggestion
 * Copyright (C) 2023 SUSE LLC
```

**Review by jecluis — DISMISSED** [review]
would love to do some manual testing before merging, but the code looks good

**Comment by irq0** [comment]
Last push fixes the previous formatting error

**Pull request merged**